packages/synthesizer/__tests__/euclidean.sty

```diff
@@ -138,6 +147,110 @@ with Point p; Point q {
     --  e.icon above const.plane
 }
 
+Segment s, t
+where EqualLength(s, t) {
+  encourage equal(vdist(s.icon.start, s.icon.end), vdist(t.icon.start, t.icon.end))
+}
+
+Segment s, t 
+where EqualLengthMarker(s, t) {
+  s.tick = Line {
+    start : midpointOffset(s.icon, 10.)
+    end : midpointOffset(s.icon, -10.)
+    color : Colors.black
+    thickness : 2.0
+    stroke : "none"
+    style : "solid"
+  }
+  t.tick = Line {
+    start : midpointOffset(t.icon, 10.)
+    end : midpointOffset(t.icon, -10.)
+    color : Colors.black
+    thickness : 2.0
+    stroke : "none"
+    style : "solid"
+  }
+  s.tick above s.icon
+  t.tick above t.icon
+}
+
+Angle a, b
+where EqualAngleMarker1(a, b) {
+  --find points from p->q, then q->r for each vector. draw vectors for each
+  startA = angleMarker(a.q, a.p, a.radius)
+  endA = angleMarker(a.q, a.r, a.radius)
+  sweepA = arcSweepFlag(a.q, startA, endA)
+
+  startB = angleMarker(b.q, b.p, b.radius)
+  endB = angleMarker(b.q, b.r, b.radius)
+  sweepB = arcSweepFlag(b.q, startB, endB)
+
+  override a.angleMark = Path {
+    pathData : arc("open", startA, endA, (a.radius, a.radius), 0, 0, sweepA)
+   strokeWidth : 2.0
+    color : Colors.black
+    fill: Colors.none
+ }
+  override b.angleMark = Path {
+    pathData : arc("open", startB, endB, (b.radius, b.radius), 0, 0, sweepB)
+    strokeWidth : 2.0
+    color : Colors.black
+    fill: Colors.none
+ }
+}
+
+Angle a, b
+where EqualAngleMarker2(a, b) {
+  startA = angleMarker(a.q, a.p, a.radius)
+  endA = angleMarker(a.q, a.r, a.radius)
+  sweepA = arcSweepFlag(a.q, startA, endA)
+
+  startB = angleMarker(b.q, b.p, b.radius)
+  endB = angleMarker(b.q, b.r, b.radius)
+  sweepB = arcSweepFlag(b.q, startB, endB)
+
+  a.angleMark = Path {
```

Comment:
This is not a change request: it would be great if we can have some concept of grouping or an easy API to compose multiple paths. You could technically pack the two arcs into one `Path` shape by having 2 `Elem`s in the `pathData`, but since there's no native syntax or functions for constructing complex paths, @hsharriman has to repeatedly declare multiple `Path`s for this.